app/models/manageiq/providers/autosde/storage_manager/autosde_client.rb

```diff
@@ -69,6 +69,9 @@ def login
     opts = {:login => true}
     data = self.AuthenticationApi.token_auth_post(auth_request, opts)
     @token = data.token
+  rescue AutosdeOpenapiClient::ApiError => e
+    error_data = JSON.parse(e.response_body)
+    raise AUTH_ERRR_MSG + error_data['detail']['non_field_errors'][0]
```

Comment:
We could code this a little more defensively just in case `detail` and/or `non_field_errors` are empty, e.g.:
```ruby
raise AUTH_ERRR_MSG + error_data.dig("detail", "non_field_errors")&.first
```

But this can be tackled in a follow-up.

Reply:
Also, was `AUTH_ERRR_MSG` intentional or should it be `AUTH_ERROR_MSG`?

Reply:
And as a project MIQ prefers string interpolation (aka `"#{AUTH_ERRR_MSG}#{error_data.dig("detail", "non_field_errors")&.first}"`) over string addition like you have here.  I think rubocop didn't warn about it because it hasn't picked up on the fact that `AUTH_ERRR_MSG` is a string